### PR TITLE
Feature/sh01 test

### PR DESCRIPTION
**Pull request by danfalc**
*** some accessibility checks are commented out so this test can be run in its entirety. These will be uncommented before being merged ***

### CODE REVIEW

File: cypress/support/page_objects/generic/ShareCapital.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import BasePage from "./BasePage";
```
Comment (rsuller) [inline cypress/support/page_objects/generic/ShareCapital.js]: Eslint picking this up;
```
  5:7  warning  'invalidCharacter' is assigned a value but never used  no-unused-vars
  6:7  warning  'addressPage' is assigned a value but never used       no-unused-vars

```

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        // Check Allotment of shares pre-filing screen
+        cy.accessibilityCheck();
+        allotmentOfSharesPreFilingPage.startShareAllotment();
+
+    })
+    it('SH01 Sucessful Submission and error validation', () => {
+        // Allotment dates tab
+        //Proceed without selecting a date then check accesibility of displayed error messages
+        allotmentOfSharesPage.goToAllotmentOfSharesTab();
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: Comment in or remove?

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+        // Allotment of shares tab 
+        // Add allotment page
+        allotmentOfSharesPage.addAllotmentLink();
+        //Expand all fields, select incorrect currency and leave all other fields blank
+        addAllotmentPage.expandAll()
+            .selectIncorrectCurrency()
+            .submitForm();
+        //check accessibility of section error messages
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: Comment in or remove?

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        addAllotmentPage.selectShareClass("A Ordinary");
+        cy.accessibilityCheck();
+        addAllotmentPage.enterNumberOfSharesAndNominalValue(1, 1);
+        cy.accessibilityCheck();
+        addAllotmentPage.enterSharePayments(1, 0);
+        cy.accessibilityCheck();
+        addAllotmentPage.addAllotment();
+
+        //Check the accessibility of the Allotment of shares tab with the newly added allotment displayed
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: Comment in or remove?

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        allotmentOfSharesPage.goToStatementOfCapitalTab();
+
+        // Statement of Capital tab
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: Comment in or remove?

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        updateOrAmendCapitalPage.enterInvalidInformationInFieldsAndCheck();
+        //Enter correct information, checking each section for accessibility violations.
+        updateOrAmendCapitalPage.amendClassOfShare("A Ordinary");
+        cy.accessibilityCheck();
+        updateOrAmendCapitalPage.editNumberOfSharesAndNominalValue("1");
+        cy.accessibilityCheck();
+        updateOrAmendCapitalPage.enterPrescribedParticulars("Amend capital test " + Cypress.moment());
+        cy.accessibilityCheck();
+        updateOrAmendCapitalPage.submitForm();
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: Comment in or remove?

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+        // Add new share class to this currency page
+        //submit without selections to fire section errors and check accessibility
+        allotmentOfSharesPage.addNewShareClassToCurrency();
+        addNewShareClassToCurrencyPage.expandAll()
+        .addShareClass();
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: Comment in or remove?

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        //return to previous page to continue submission
+        addNewShareClassToCurrencyPage.cancelAddShareClass();
+
+        // Add new currency and share class page
+        //submit without selections to fire section errors and check accessibility
+        allotmentOfSharesPage.addNewCurrencyAndShareClass();
+        addNewCurrencyAndShareClassPage.expandAll()
+        .selectIncorrectCurrency()
+        .addCurrencyAndShareClass();
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: Comment in or remove?

File: cypress/integration/share_capital/AllotmentOfShares.spec.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+        cy.accessibilityCheck();
+
+        //Valid total aggregate amount unpaid
+        allotmentOfSharesPage.enterTotalAggregateAmountUnpaid(0);
+
+        // Submit and confirm submission
+        allotmentOfSharesPage.submitShareAllotment();
+        submissionConfirmationPage.confirmHeadingContains('Confirmation of Submission')
+        cy.accessibilityCheck();
+
```
Comment (rsuller) [inline cypress/integration/share_capital/AllotmentOfShares.spec.js]: A very busy script but guess it's due to the nature of the form and it's set up?

File: cypress/support/page_objects/AddAllotmentPage.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (rsuller) [inline cypress/support/page_objects/AddAllotmentPage.js]: Remove the white space?

File: cypress/support/page_objects/AddNewCurrencyAndShareClassPage.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (rsuller) [inline cypress/support/page_objects/AddNewCurrencyAndShareClassPage.js]: Remove the white space?

File: cypress/support/page_objects/AddNewShareClassToCurrencyPage.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (rsuller) [inline cypress/support/page_objects/AddNewShareClassToCurrencyPage.js]: Is the white space being added by VS code?

File: cypress/support/page_objects/UpdateOrAmendCapitalPage.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (rsuller) [inline cypress/support/page_objects/UpdateOrAmendCapitalPage.js]: White space.

File: cypress/support/page_objects/UpdateOrAmendCapitalPage.js [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+import ShareCapital from "./generic/ShareCapital";
+
+const invalidCharacter = "`";
```
Comment (rsuller) [inline cypress/support/page_objects/UpdateOrAmendCapitalPage.js]: Is this starting to get repeated in other pages? Worth moving to a fixture file?

File: cypress/support/page_objects/UpdateOrAmendCapitalPage.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+    enterPrescribedParticulars(particulars) {
+        cy.get('#prescribed-particulars-container-change').click();
+        cy.accessibilityCheck();
+        cy.get('#PrescribedParticulars').clear().type(particulars)
+        cy.get('#prescribed-particulars-container-continue').click();
+    }
+
+    enterInvalidInformationInFieldsAndCheck() {
+        this.amendClassOfShare('Please select');
+        //cy.accessibilityCheck();
```
Comment (rsuller) [inline cypress/support/page_objects/UpdateOrAmendCapitalPage.js]: Accessibility checks required?

**Comment by rsuller** [comment]
> *** some accessibility checks are commented out so this test can be run in its entirety. These will be uncommented before being merged ***

I love how I have 100% ignored this 🤦‍♂

**Comment by danfalc** [comment]
> > *** some accessibility checks are commented out so this test can be run in its entirety. These will be uncommented before being merged ***
> 
> I love how I have 100% ignored this 🤦‍♂

That is absolutely fine by me. Pointed out everywhere I need to make changes saves me a job 👍

File: cypress/support/page_objects/UpdateOrAmendCapitalPage.js (hunk 14 above)
Comment (danfalc) [reply cypress/support/page_objects/UpdateOrAmendCapitalPage.js]: It is a lot more widespread than I thought. Should probably be done in a separate refactoring PR

**Comment by danfalc** [comment]
All comments have been addressed

File: cypress/support/page_objects/UpdateOrAmendCapitalPage.js (hunk 14 above)
Comment (rsuller) [reply cypress/support/page_objects/UpdateOrAmendCapitalPage.js]: 👍

File: cypress/integration/share_capital/AllotmentOfShares.spec.js (hunk 9 above)
Comment (danfalc) [reply cypress/integration/share_capital/AllotmentOfShares.spec.js]: Yeah there's alot involved and all tabs hang from this screen. I've tried to explain with comments

**Pull request merged**